apps/prairielearn/src/pages/instructorAssessmentManualGrading/assessmentQuestion/AssessmentQuestionManualGrading.tsx

```diff
@@ -0,0 +1,189 @@
+  courseStaff: StaffUser[];
+  aiGradingStats: AiGradingGeneralStats | null;
+  numOpenInstances: number;
+  search: string;
+  isDevMode: boolean;
+  questionTitle: string;
+  questionNumber: number | null;
+}
+
+type AssessmentQuestionManualGradingInnerProps = Omit<
+  AssessmentQuestionManualGradingProps,
+  'search' | 'isDevMode'
+>;
+
+function AssessmentQuestionManualGradingInner({
+  hasCourseInstancePermissionEdit,
+  instanceQuestionsInfo,
+  course,
+  courseInstance,
+  urlPrefix,
+  csrfToken,
+  assessment,
+  assessmentQuestion,
+  questionQid,
+  aiGradingEnabled,
+  initialAiGradingMode,
+  rubricData,
+  instanceQuestionGroups,
+  courseStaff,
+  aiGradingStats,
+  numOpenInstances,
+  questionTitle,
+  questionNumber,
+}: AssessmentQuestionManualGradingInnerProps) {
+  const [groupInfoModalState, setGroupInfoModalState] = useState<GroupInfoModalState>(null);
+  const [showConflictModal, setShowConflictModal] = useState(false);
+  const [conflictDetailsUrl, setConflictDetailsUrl] = useState('/');
+
+  const handleShowConflictModal = (url: string) => {
+    setConflictDetailsUrl(url);
+    setShowConflictModal(true);
+  };
+
+  const [aiGradingMode, setAiGradingMode] = useState(initialAiGradingMode);
+
+  const { groupSubmissionMutation, toggleAiGradingModeMutation, ...mutations } =
+    useManualGradingActions({
+      csrfToken,
+      courseInstanceId: courseInstance.id,
+    });
+
+  return (
+    <>
+      {toggleAiGradingModeMutation.isError && (
+        <Alert
+          variant="danger"
+          class="mb-3"
+          dismissible
+          onClose={() => toggleAiGradingModeMutation.reset()}
+        >
+          <strong>Error:</strong> {toggleAiGradingModeMutation.error.message}
+        </Alert>
+      )}
+      <div class="d-flex flex-row justify-content-between align-items-center mb-3 gap-2">
+        <nav aria-label="breadcrumb">
+          <ol class="breadcrumb mb-0">
+            <li class="breadcrumb-item">
+              <a href={`${urlPrefix}/assessment/${assessment.id}/manual_grading`}>Manual grading</a>
+            </li>
+            <li class="breadcrumb-item active" aria-current="page">
+              Question {questionNumber != null ? questionNumber : ''}. {questionTitle}
+            </li>
+          </ol>
+        </nav>
+        {aiGradingEnabled && (
+          <div class="card px-3 py-2 mb-0">
+            <div class="form-check form-switch mb-0">
+              <input
+                class="form-check-input"
+                type="checkbox"
+                role="switch"
+                id="switchCheckDefault"
+                checked={aiGradingMode}
+                disabled={toggleAiGradingModeMutation.isPending}
+                onChange={() =>
+                  toggleAiGradingModeMutation.mutate(undefined, {
+                    onSuccess: () => {
+                      setAiGradingMode((prev) => !prev);
+                    },
+                  })
```

Comment:
It seems like having the action be "toggle" rather than "set" means we run the risk of the client and server getting out of sync if two people are viewing the page at the same time: